integrations/transportation/wizzair.js

```diff
@@ -0,0 +1,187 @@
+import moment from 'moment';
+import request from 'superagent';
+import { ACTIVITY_TYPE_TRANSPORTATION, TRANSPORTATION_MODE_PLANE } from '../../definitions';
+import { HTTPError, AuthenticationError, ValidationError } from '../utils/errors';
+
+const API_VERSION_URL = 'https://wizzair.com/static/metadata.json';
+const agent = request.agent();
+const HISTORY_API_FETCH_LIMIT = 5;
+
+async function getApiVersionUrl() {
+  return fetch(API_VERSION_URL).then(response => response.text());
+}
+
+let API_URL;
+let BASE_URL;
+let LOGIN_URL;
+let ITINERARY_URL;
+let BOOKINGS_URL;
+
+async function logIn(username, password) {
+  // urls have to be assigned here as they're asynchronous and have to be awaited
+  await Promise.all(
+    API_URL = await getApiVersionUrl(),
+    BASE_URL = JSON.parse(API_URL.trim()).apiUrl,
+    LOGIN_URL = `${await BASE_URL}/customer/login`,
+    ITINERARY_URL = `${await BASE_URL}/booking/itinerary`,
+    BOOKINGS_URL = `${await BASE_URL}/customer/mybookings`,
+  );
+
+  const res = await agent
+    .post(LOGIN_URL)
+    .type('application/json')
+    .set('Accept', '*/*')
+    .send({
+      username,
+      password,
+      isAgencyLogin: false,
+      captchaResponse: '',
+      languageCode: 'en-gb',
+    });
+
+  if (!res.ok) {
+    throw new AuthenticationError(`Error while logging in. ${res.body}`);
```

Comment:
```suggestion
    const text = await res.text();
    throw new HTTPError(text, res.status);
```

Reply:
I noticed something about reporting errors this way. 'res.text()' returns a syntax error because it's not a function, so I think 'res.text' should be there instead. Then, as VS code warnings show, it doesn't have to be awaited anymore. I updated the code this way, let me know if it's ok! 🙆

Reply:
Ah, you are right. Good catch!